### PR TITLE
Bugfix: Remove TotemT2 Envelope Volume

### PR DESCRIPTION
**Pull request by ianna**
* Add a forward:TotemT2 missing volume which is used in other files: ionpump.xml and castor.xml

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

The scenario is orthogonal to the production workflows and is not used in the relval tests.

**Comment by civanch** [comment]
@ianna , Totem2 has been removed for 2018 run physically, Castor not yet installed, it will be installed in November. It was discussed in the beginning of the year, that Totem2 is taken out but Castor geometry  will be left, because Castor cannot affect central detector. If I understand correctly the description of this PR, then it would be better to close this PR.

**Comment by civanch** [comment]
@ianna , then, please, change the title and description, it is not return back Totem station 2.

**Comment by ianna** [comment]
@civanch - the description is accurate since the error is a missing Totem2 volume defined in the forward.xml file. It took me a while to trace it, so it is better to keep it for future reference.

**Comment by civanch** [comment]
@ianna , now I understand! May be enough to add "envelope" in the description

**Comment by kpedro88** [comment]
+upgrade

**Comment by ianna** [comment]
The latest commits added to remove the following warnings related to an incorrect definition of the Totem2 envelope volume: 
```
WARNING: MCParticlePairFilter : size of some vectors not matching with 2!!
Cannot create Polycone with rInner > rOuter for the same Z
         rInner > rOuter for the same Z !
         rMin[2] = 3200 -- rMax[2] = 250Cannot create a Polycone with no 
contiguous segments.
                 Segments are not contiguous !
                 rMin[1] = 42 -- rMax[2] = 250
                 rMin[2] = 3200 -- rMax[1] = 250%MSG-w 
SimG4CoreApplication:  (NoModuleName) 19-Sep-2018 13:48:09 CEST pre-events

-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomMgt0001
       issued by : G4UPolycone::BoundingLimits()
Inconsistency in bounding boxes for solid: TotemT2 !
BBox min: wrapper = (-3200,-3200,13381) solid = (-250, -250, 13381)
BBox max: wrapper = (3200,3200,15350) solid = (250, 250, 15350)*** This 
is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------

and some other warnings about TotemT2...

```

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

The scenario is orthogonal to the production workflows and is not used in the relval tests.

**Comment by kpedro88** [comment]
+upgrade

File: Geometry/CMSCommonData/data/cms/2018/v1/cms.xml [hunk 1]
```diff
@@ -56,19 +56,18 @@
...
- <Constant name="TotemBeamR1"  value="[MuonBeamR1]"/>
  <Constant name="TotemBeamR2"  value="12.15*cm"/>
  <Constant name="TotemBeamR3"  value="12.20*cm"/>
- <Constant name="TotemBeamR4"  value="4.20*cm"/>
- <Constant name="TotemBeamR5"  value="320.0*cm"/>
+ <Constant name="TotemBeamR4"  value="3.675*cm"/>
+ <Constant name="TotemBeamR5"  value="3.575*cm"/>
+ <Constant name="TotemBeamZ1"  value="7.9500*m"/>
+ <Constant name="TotemBeamZ2"  value="13.381*m"/>
+ <Constant name="TotemBeamZ3"  value="13.439*m"/>
```
Comment (fabiocos) [inline Geometry/CMSCommonData/data/cms/2018/v1/cms.xml]: @ianna @civanch does this and other lines imply that in 10_2_X we are using a wrong geometry description for the Fall18 production?

File: Geometry/CMSCommonData/python/cmsExtendedGeometryDetailedCavern2018XML_cfi.py [hunk 2]
```diff
@@ -239,10 +239,10 @@
         'Geometry/MuonCommonData/data/mfshield/2017/v1/mfshield.xml',
         'Geometry/MuonCommonData/data/gemf/TDR_BaseLine/gemf.xml',
         'Geometry/MuonCommonData/data/gem11/2017/v2/gem11.xml',
+        'Geometry/ForwardCommonData/data/forward/2018/v1/forward.xml',
```
Comment (fabiocos) [inline Geometry/CMSCommonData/python/cmsExtendedGeometryDetailedCavern2018XML_cfi.py]: @ianna @civanch the addition of CASTOR should be relevant only for HI data, right?

File: Geometry/CMSCommonData/python/cmsExtendedGeometryDetailedCavern2018XML_cfi.py [hunk 3]
```diff
@@ -239,10 +239,10 @@
...
         'Geometry/ForwardCommonData/data/forwardshield/2017/v1/forwardshield.xml',
         'Geometry/ForwardCommonData/data/brmrotations.xml',
         'Geometry/ForwardCommonData/data/PostLS2/brm.xml',
-        'Geometry/ForwardCommonData/data/ionpump.xml',
```
Comment (fabiocos) [inline Geometry/CMSCommonData/python/cmsExtendedGeometryDetailedCavern2018XML_cfi.py]: @ianna @civanch is this material that will be removed now?

**Comment by fabiocos** [comment]
@ianna although this is for 10_3_X (to be used for HI data taking), could you please clarify the possible implications of this PR for 10_2_X ongoing SIM production? Are we using a wrong geometry configuration there?

File: Geometry/CMSCommonData/python/cmsExtendedGeometryDetailedCavern2018XML_cfi.py (hunk 2 above)
Comment (ianna) [reply Geometry/CMSCommonData/python/cmsExtendedGeometryDetailedCavern2018XML_cfi.py]: Yes. Unfortunately, there are some modules which ask for Castor hits in any case. Given that the production stops when they are not available, the easiest way was to keep it in the geometry description. The alternative would be to customize the producers, but it's not in my domain.

**Comment by ianna** [comment]
@fabiocos - If the 10_2_X production is ongoing, there should be no problem. This PR is for an alternative scenario.

**Comment by civanch** [comment]
@fabiocos , we discuss at spring, that  TOTEM1,2 detectors are physically dismantled forever and Castor should be installed in November, so we take away TOTEM from production of 2018 but leave Castor in in order to have uniform simulation for whole year. Physically any particle entering Castor will never reflected back in our simulation, CPU overhead due to Castor is minimal.  These were arguments to keep Castor.

**Comment by fabiocos** [comment]
+1

**Pull request merged**